File: Languages/Zig/Practice/Pointers/square.zig
const expect = @import("std").testing.expect;

test "square" {
    var x: i32 = 10;
    
    const before = x;

    square(&x);

    try expect(x == before * before);
}

fn square(num: *i32) void {
    num.* *= num.*;
}

File: languages/c/practice/pointers/square.c
#include <stdio.h>

void square(int *num);

int main(void)
{
    int num = 10;

    square(&num);
    
    printf("Result: %d\n", num);

    return 0;
}

void square(int *num)
{
    *num *= *num;
}